File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
class Solution
{
    public int kthSmallest(TreeNode root, int k)
    {
        Deque<TreeNode> stack = new ArrayDeque<>();
        TreeNode cur = root;

        while (!stack.isEmpty() || cur != null)
        {
            while (cur != null)
            {
                stack.push(cur);
                cur = cur.left;
            }

            cur = stack.pop();
            k--;

            if (k == 0)
            {
                return cur.val;
            }

            cur = cur.right;
        }
        return -1;
    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        #INORDER TRAVERSAL ITERATIVE METHOD

        res = []
        stack = []
        cur = root 

        while stack or cur:
            while cur:
                stack.append(cur)
                cur = cur.left
            
            cur = stack.pop()
            res.append(cur.val)
            cur = cur.right
        return res[k-1]
            



        #INORDER TRAVERSAL 
        res = []

        def dfs(node):

            # BASE CASE 
            if not node:
                return 

            dfs(node.left)
            res.append(node.val)
            dfs(node.right)

        dfs(root)
        return res[k- 1]

        
      
